core-java-modules/core-java-optional/src/test/java/com/baeldung/optionaluses/UsesForOptionalUnitTest.java

```diff
@@ -50,15 +51,38 @@ public void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUs
     }
 
     @Test
-    public void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUsingOptional_thenNameShouldBeUpperCased() {
+    void givenExistentUserId_whenFoundUserWithNameStartingWithMInRepositoryUsingOptional_thenNameShouldBeUpperCased() {
 
         UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
 
         String upperCasedName = userRepositoryWithOptional.findById("2")
-          .filter(u -> u.getName().startsWith("M"))
-          .map(u -> u.getName().toUpperCase())
-          .orElse("");
+            .filter(u -> u.getName().startsWith("M"))
+            .map(u -> u.getName().toUpperCase())
+            .orElse("");
 
         assertEquals("MARIA", upperCasedName);
     }
+
+    @Test
+    void givenExistentUserId_whenSearchForUser_thenThrowException() {
+
+        final UserRepositoryWithOptional userRepositoryWithOptional = new UserRepositoryWithOptional();
+        String existentUserId = "2";
+
+        assertThrows(UserFoundException.class, () -> {
+            userRepositoryWithOptional.throwExceptionWhenUserIsPresent(existentUserId);
+        });
```

Comment:
I didn't wanted to be redundant but I was talking about all the lambda statement.

Reply:
Oops, my bad. Done